hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,85 @@
+        param_dict = ParameterDict(
+            test_particle_type=str,
+            num_samples=int
+        )
+        param_dict.update(
+            dict(test_particle_type=test_particle_type,
+                 num_samples=num_samples))
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be an HPMC integrator.")
+
+        # Extract 'Shape' from '<hoomd.hpmc.integrate.Shape object>'
+        integrator_name = str(integrator).split()[0].split('.')[-1]
+        try:
+            if isinstance(self._simulation.device, hoomd.device.CPU):
+                cpp_cls = getattr(_hpmc, 'ComputeFreeVolume' + integrator_name)
             else:
-                hoomd.context.current.device.cpp_msg.error("compute.free_volume: Unsupported integrator.\n");
-                raise RuntimeError("Error initializing compute.free_volume");
-        else:
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.ComputeFreeVolumeGPUSphere;
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolygon;
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.ComputeFreeVolumeGPUSimplePolygon;
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolyhedron;
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.ComputeFreeVolumeGPUSpheropolyhedron;
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.ComputeFreeVolumeGPUEllipsoid;
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls =_hpmc.ComputeFreeVolumeGPUSpheropolygon;
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls =_hpmc.ComputeFreeVolumeGPUFacetedEllipsoid;
-            elif isinstance(mc, integrate.polyhedron):
-                cls =_hpmc.ComputeFreeVolumeGPUPolyhedron;
-            elif isinstance(mc, integrate.sphinx):
-                cls =_hpmc.ComputeFreeVolumeGPUSphinx;
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ComputeFreeVolumeGPUSphereUnion;
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ComputeFreeVolumeGPUFacetedEllipsoidUnion;
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolyhedronUnion;
-            else:
-                hoomd.context.current.device.cpp_msg.error("compute.free_volume: Unsupported integrator.\n");
-                raise RuntimeError("Error initializing compute.free_volume");
+                cpp_cls = getattr(_hpmc, 'ComputeFreeVolume' + integrator_name + 'GPU')
```

Comment:
It seems the appropriate class name would be `'ComputeFreeVolumeGPU' + integrator_name`.

Reply:
Should I change this in the pybind export in all of the `module.cc` files? Because for example `module_convex_polyhedron.cc` has this:

`export_ComputeFreeVolumeGPU< ShapeConvexPolyhedron >(m, "ComputeFreeVolumeConvexPolyhedronGPU");`

Reply:
Let's make an issue to clean this up later to avoid conflicts with #659 which rewrites how all of these module files.